{
  try {
    if (callback != null)     callback.failed(x);
  }
 catch (  Exception xx) {
    LOG.info("Exception while notifying callback " + callback,xx);
  }
catch (  Error xx) {
    LOG.info("Exception while notifying callback " + callback,xx);
    throw xx;
  }
}
